src/lib/routing-tables.js

```diff
@@ -0,0 +1,70 @@
+'use strict'
+const routing = require('five-bells-routing')
+
+/**
+ * Maintain two sets of `routing.RoutingTables`.
+ *
+ * The "local" tables consist of the raw local routes, joined to the
+ * remote (shifted) routes.
+ *
+ * The "public" tables consist of the local shifted routes, joined to the
+ * remote (shifted) routes.
+ * Before joining, the local routes are shifted down by 1/10^destination_ledger_scale
+ * to pessimistically account for rounding errors.
+ *
+ * The "local" tables are used for `findBestHopFor*`.
+ * The "public" tables are broadcast to adjacent connectors.
```

Comment:
How about:


> When routing payments across multiple ledgers, each hop will round the amounts according to the precision of the ledger. These rounding errors can accumulate to the point where a connector later in a path may receive an amount that is less than the rate they are willing to accept.
> 
> Connectors broadcast slightly pessimistic rate curves in order to compensate for rounding errors across multi-hop routes. They shift the curve by the maximum amount that could be lost due to rounding errors, so that even if the incoming amount is rounded down by that amount the resulting value will still match the connector's minimum rate.
> 
> Connectors use unshifted rate curves locally to determine whether an incoming payment request matches their minimum rate. When applying this rate, they round in their favor to insure that they never accept a payment that is lower than their rate.
> 
> This class maintains two sets of `routing.RoutingTables`: the "local" tables are the optimistic ones and the "public" tables are the pessimistic ones.
> 
> The "local" tables consist of the raw local routes, joined to adjacent connector's (public, shifted) routes. They are used for `findBestHopFor*`, the results of which must be rounded in the connector's own favor to ensure that they don't lose money.
> 
> The "public" tables consist of the local shifted routes, joined to the remote (shifted) routes. Before joining, the local routes are shifted down by 1/10^destination_ledger_scale to account for rounding errors. Their curves are broadcast to adjacent connectors.